node/flatpak-node-generator.py

```diff
@@ -606,6 +606,16 @@ def add_script_source(self, commands: List[str], destination: Path) -> None:
         source = {'type': 'script', 'commands': tuple(commands)}
         self._add_source_with_destination(source, destination, is_dir=False)
 
+    def add_shell_source(self,
+                         commands: List[str],
+                         destination: Optional[Path] = None,
+                         only_arches: Optional[List[str]] = None):
+        source = {'type': 'shell', 'commands': tuple(commands)}
```

Comment:
We already have `add_command` to any commands needed once generation of everything else is complete (in `_finalize`), is there a particular reason for a separate add_shell_source? (esp. since the shell source doesn't actually take destination or arches, those will just be ignored, slightly surprised flatpak-builder didn't warn)

Reply:
Yes, that's because of only_arches. Without it, we'll create a plenty of irrelevant broken symlinks. They're probably harmless, but it feels a bit messy.

Reply:
Hmm, that's true. For some context, the reason scripts are all grouped together is just because the npm generator generates *so many*, it seriously kills performance. A cleaner solution might be to try to group scripts together but...that's a bit overkill for now given that there's only only one user... For now it should probably at least be `add_shell_source_slow` or similar w/ a warning comment to ideally not use it :sweat_smile:

Reply:
Added method description suggesting to use `add_command()` instead.